kart/annotations/db.py

```diff
@@ -0,0 +1,51 @@
+import contextlib
+import json
+import threading
+from sqlalchemy import Column, Integer, Text
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from kart.sqlalchemy.create_engine import sqlite_engine
+
+Base = declarative_base()
+
+
+class KartAnnotation(Base):
+    __tablename__ = "kart_annotations"
+    id = Column(Integer, nullable=False, primary_key=True)
+    # TODO: look into what indexes we should have here
```

Comment:
presumably a multi-column index... depends on what queries we'll likely need for _multiple objects_?

1. (`object_type`, `object_id`, `annotation_type`) would allow efficiently enumerating all annotations for object types, or a specific object.
2. (`annotation_type`, `object_type`, `object_id`) would allow enumerating all objects that have a specific annotation.

Reply:
really unless we're expecting _at least_ hundreds of thousands of these, any indexes at all might be completely unnecessary.

So far annotations don't get looked up for multiple objects at once. A future enhancement for `log` or `build-annotations` might change to doing that though.

All lookups at present are based on all three of `(object_type, annotation_type, object_id)`

Reply:
A crude representative sqlite test has the index as 3 times quicker even for 500 rows. Block cache obviously makes it go away fairly quickly, but considering this stuff will almost never be in the cache seems like minimal disk hits are key?

Reply:
ok

`(annotation_type, object_type, object_id)` seems like the best choice to me.